db/migrate/20160415134454_add_ipv6_to_hosts.rb

```diff
@@ -0,0 +1,15 @@
+class AddIpv6ToHosts < ActiveRecord::Migration
+  def self.up
+    change_column :nics, :ip, :string, :limit => 15
+    add_column :nics, :ip6, :string, :limit => 45
+    add_column :hostgroups, :subnet6_id, :integer
+    add_column :nics, :subnet6_id, :integer
+  end
+
+  def self.down
+    change_column :nics, :ip, :string
+    remove_column :nics, :ip6
+    remove_column :hostgroups, :subnet6_id
+    remove_column :nics, :subnet6_id
+  end
+end
```

Comment:
Is it time to introduce indicies for both `:ip` and `:ip6` columns? We search during unattended installations mainly, then users do search via scoped_search. Opinion?

```
Searching 2828 files for "by_ip"

/home/lzap/work/foreman/app/controllers/unattended_controller.rb:
   77  
   78    def get_host_details
   79:     @host = find_host_by_spoof || find_host_by_token || find_host_by_ip_or_mac
   80      unless @host
   81        logger.info "#{controller_name}: unable to find a host that matches the request from #{request.env['REMOTE_ADDR']}"
   ..
   95  
   96    def find_host_by_spoof
   97:     host = Nic::Base.primary.find_by_ip(params.delete('spoof')).try(:host) if params['spoof'].present?
   98      host ||= Host.find_by_name(params.delete('hostname')) if params['hostname'].present?
   99      @spoof = host.present?
  ...
  111    end
  112  
  113:   def find_host_by_ip_or_mac
  114      # try to find host based on our client ip address
  115      ip = ip_from_request_env

/home/lzap/work/foreman/test/functional/unattended_controller_test.rb:
  174      get :built
  175      assert_response :created
  176:     host = Nic::Base.primary.find_by_ip(@ub_host.ip)
  177      assert_equal host.build,false
  178    end

/home/lzap/work/foreman/test/unit/lookup_value_test.rb:
   39      # Host.authorized(:view_hosts, Host) returns only hosts(:one)
   40      user = users(:one)
   41:     role = FactoryGirl.create(:role, :name => 'user_view_host_by_ip')
   42      FactoryGirl.create(:filter, :role => role, :permissions => [Permission.find_by_name(:view_hosts)], :search => "name = #{@host1.name}")
   43      # Todo, restore the ip test variant once our scoped-search works with host.ip again

/home/lzap/work/foreman_bootdisk/app/controllers/concerns/foreman_bootdisk/unattended_controller_ext.rb:
    3  
    4    included do
    5:     alias_method_chain :find_host_by_ip_or_mac, :param_mac
    6    end
    7  
    8:   def find_host_by_ip_or_mac_with_param_mac
    9      request.env['HTTP_X_RHN_PROVISIONING_MAC_0'] = "unknown #{params['mac']}" unless request.env.has_key?('HTTP_X_RHN_PROVISIONING_MAC_0') || params['mac'].nil?
   10:     find_host_by_ip_or_mac_without_param_mac
   11    end
   12  end

8 matches across 4 files
```